pkg/opentelemetry/output.go

```diff
@@ -150,31 +145,43 @@ func (o *Output) dispatch(entry metrics.Sample) error {
 	ctx := context.Background()
 	name := normalizeMetricName(o.config, entry.Metric.Name)
 
+	attributeSet := newAttributeSet(entry.Tags)
+	attributeSetOpt := otelMetric.WithAttributeSet(attributeSet)
+
 	switch entry.Metric.Type {
 	case metrics.Counter:
-		counter, err := o.getOrCreateCounter(name)
+		counter, err := o.metricsRegistry.getOrCreateCounter(name)
 		if err != nil {
 			return err
 		}
 
-		counter.Add(ctx, entry.Value, otelMetric.WithAttributes(MapTagSet(entry.Tags)...))
+		counter.Add(ctx, entry.Value, attributeSetOpt)
 	case metrics.Gauge:
-		gauge, err := o.getOrCreateUpDownCounter(name)
+		gauge, err := o.metricsRegistry.getOrCreateGauge(name)
 		if err != nil {
 			return err
 		}
 
-		gauge.Add(ctx, entry.Value, otelMetric.WithAttributes(MapTagSet(entry.Tags)...))
+		gauge.Set(entry.Value, attributeSet)
 	case metrics.Trend:
-		trend, err := o.getOrCreateHistogram(name)
+		trend, err := o.metricsRegistry.getOrCreateHistogram(name)
+		if err != nil {
+			return err
+		}
+
+		trend.Record(ctx, entry.Value, attributeSetOpt)
+	case metrics.Rate:
+		nonZero, total, err := o.metricsRegistry.getOrCreateCountersForRate(name)
 		if err != nil {
 			return err
 		}
 
-		trend.Record(ctx, entry.Value, otelMetric.WithAttributes(MapTagSet(entry.Tags)...))
+		if entry.Value > 0 {
```

Comment:
This should probably be `!=` to match both the name and the in k6 sink

https://github.com/grafana/k6/blob/44d784007acf61be7215d117ce088f2bf4e69585/metrics/sink.go#L212

```suggestion
		if entry.Value != 0 {
```